pennylane/transforms/commutation_dag.py

```diff
@@ -0,0 +1,1049 @@
+    "DisplacementEmbedding",
+    "SqueezingEmbedding",
+]
+non_commuting_operations = [
+    "ArbitraryStatePreparation",
+    "BasisStatePreparation",
+    "MottonenStatePreparation",
+    "QubitCarry",
+    "QubitSum",
+    "SingleExcitation",
+    "SingleExcitationMinus",
+    "SingleExcitationPlus",
+    "DoubleExcitation",
+    "DoubleExcitationPlus",
+    "DoubleExcitationMinus",
+    "BasicEntanglerLayers",
+    "GateFabric",
+    "ParticleConservingU1",
+    "ParticleConservingU2",
+    "RandomLayers",
+    "SimplifiedTwoDesign",
+    "StronglyEntanglingLayers",
+    "AllSinglesDoubles",
+    "FermionicDoubleExcitation",
+    "FermionicSingleExcitation",
+    "Grover",
+    "kUpCCGSD",
+    "Permute",
+    "QFT",
+    "QuantumMonteCarlo",
+    "QuantumPhaseEstimation",
+    "UCCSD",
+    "MPS",
+    "TTN",
+    "AmplitudeEmbedding",
+    "AngleEmbedding",
+    "BasisEmbedding",
+    "IQPEmbedding",
+    "QAOAEmbedding",
+]
+
+
+def is_commuting(operation1, operation2):
+    r"""Check if two operations are commuting. A lookup table is used to check the commutation between the
+    controlled, targetted part of operation 1 with the controlled, targetted part of operation 2. It supports
+    most PennyLane operations that are not CV operations.
+
+    Args:
+        operation1 (.Operation): A first quantum operation.
+        operation2 (.Operation): A second quantum operation.
+
+    Returns:
+         bool: True if the operations commute, False otherwise.
+
+    **Example**
+
+    >>> qml.is_commuting(qml.PauliX(wires=0), qml.PauliZ(wires=0))
+    False
+    """
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-return-statements
+
+    control_base = {
+        "CNOT": "PauliX",
+        "CZ": "PauliZ",
+        "CY": "PauliY",
+        "CSWAP": "SWAP",
+        "Toffoli": "PauliX",
+        "ControlledPhaseShift": "PhaseShift",
+        "CRX": "RX",
+        "CRY": "RY",
+        "CRZ": "RZ",
+        "CRot": "Rot",
+        "MultiControlledX": "PauliX",
+        "ControlledOperation": "ControlledOperation",
+    }
+
+    if operation1.name in not_supported_operations or isinstance(
+        operation1, (qml.operation.CVOperation, qml.operation.Channel)
+    ):
+        raise qml.QuantumFunctionError(f"Operation {operation1.name} not supported.")
+
+    if operation2.name in not_supported_operations or isinstance(
+        operation2, (qml.operation.CVOperation, qml.operation.Channel)
+    ):
+        raise qml.QuantumFunctionError(f"Operation {operation2.name} not supported.")
+
+    if operation1.name == "ControlledOperation" and operation1.control_base == "MultipleTargets":
+        raise qml.QuantumFunctionError(f"{operation1.control_base} controlled is not supported.")
+
+    if operation2.name == "ControlledOperation" and operation2.control_base == "MultipleTargets":
+        raise qml.QuantumFunctionError(f"{operation2.control_base} controlled is not supported.")
```

Comment:
do these attributes need to be updated?

Reply:
@josh146 Sorry I do not understand this comment, what attributes are you talking about?

Reply:
Oh I meant `control_base`, I wasn't sure where it was defined 🤔

Reply:
Finally I only defined it for `ControlledOperation`, as the target can be different.